### PR TITLE
[DevMenu] fixing shake to open Dev Menu only work once

### PR DESCRIPTION
**Pull request by VojtechBartos**
Fixing issues https://github.com/facebook/react-native/issues/10588

This bug has been introduced during refactoring UIActionSheet to UIAlertController https://github.com/facebook/react-native/commit/d368ebfab2c389cc093f6a73e5c1021408cc6a73#diff-f78a84ff95f2bcee5a70d3accb504528

**Test plan**

- shake to open DevMenu, choose random menu item, it should close DevMenu and after another shaking it should open again

### CODE REVIEW

**Comment by wuchu** [comment]
**# Another solution**

Actually, we can use OC's runtime to fix it if you just want still use the 0.37 version.

```Object-c
// JUST FOR DEVELOPMENT
// TODO: THE NEXT RN VERSION WILL BE REMOVED.

#import "RCTDevMenu+BugFixed.h"

@implementation RCTDevMenu (BugFixed)

static void (*originalImplementationShow)(id, SEL);

static void newImplementationShow(id self, SEL _cmd) {
    [self ivar_setNilValueForName:@"_actionSheet"];
    
    if (originalImplementationShow) {
        originalImplementationShow(self, _cmd);
    }
}

+ (void)load {
    SEL selector = @selector(show);
    Method originalMethod = class_getInstanceMethod(self, selector);
    originalImplementationShow = (void *)method_getImplementation(originalMethod);
    class_replaceMethod(self, selector, (IMP) newImplementationShow, method_getTypeEncoding(originalMethod));
}

- (void) ivar_setNilValueForName:(NSString *)name {
    unsigned int count;
    Ivar *iVars = class_copyIvarList(self.class, &count);
    for (unsigned int i = 0; i < count; i++) {
        Ivar iVar = iVars[i];
        NSString *theName = [NSString stringWithUTF8String:ivar_getName(iVar)];
        if ([name isEqualToString:theName]) {
            object_setIvar(self, iVar, nil);
            break;
        }
    }
    free(iVars);
}

@end
```

Good Luck!

**Comment by VojtechBartos** [comment]
@dlowder-salesforce tests added. 

Have to slightly refactor the changes to be able to test the handlers, because i haven't found a way how to trigger UIAlertAction programatically.

What do you think?

**Comment by VojtechBartos** [comment]
@wuchu thanks for the another solution. Will use it as a workaround on my local! 👍

**Comment by wuchu** [comment]
@dlowder-salesforce 😁

**Comment by VojtechBartos** [comment]
@dlowder-salesforce ping :)

**Comment by dlowder-salesforce** [comment]
@VojtechBartos I'm fine with the changes, but someone from FB will be needed to merge your PR :)

**Comment by VojtechBartos** [comment]
@dlowder-salesforce thanks! 

@javache is there a chance you could merge this PR ? :)

**Comment by VojtechBartos** [comment]
@javache ping :)

File: Examples/UIExplorer/UIExplorerUnitTests/RCTDevMenuTests.m [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  _bridge = [[RCTBridge alloc] initWithBundleURL:[bundle URLForResource:@"TestBundle" withExtension:@"js"]
+                                  moduleProvider:nil
+                                   launchOptions:nil];
+}
+
+- (void)testShowCreatingActionSheet
+{
+  XCTAssertNil([_bridge.devMenu valueForKey:@"_actionSheet"]);
+  [_bridge.devMenu show];
+  XCTAssertNotNil([_bridge.devMenu valueForKey:@"_actionSheet"]);
```
Comment (javache) [inline Examples/UIExplorer/UIExplorerUnitTests/RCTDevMenuTests.m]: Thanks for adding tests, but testing internals like this is not very helpful. `_actionSheet` is not part of the public interface and depending in on tests will easily break.

Comment (javache) [inline Examples/UIExplorer/UIExplorerUnitTests/RCTDevMenuTests.m]: Alternatively: add a getter "isActionSheetShown" that checks if actionSheet is non-nil?

Comment (VojtechBartos) [inline Examples/UIExplorer/UIExplorerUnitTests/RCTDevMenuTests.m]: @javache make sense, will add a getter. Thanks!

**Comment by VojtechBartos** [comment]
@javache updated

**Comment by mkonicek** [comment]
@VojtechBartos Thanks for fixing this!

Tests failed on Travis, not sure if related. Restarted the build.

Circle CI tests failed with unrelated Flow errors. Can you try rebasing?

**Comment by VojtechBartos** [comment]
@mkonicek thanks for restarting the travis build, but it's failed again, not sure why, on local the iOS ones passing without any issue.

Anyway, have tried merge master locally and run the tests and it's failing, but not by this branch changes, actually [the master failing](https://travis-ci.org/facebook/react-native/builds/178352895)

Will try to do merge tomorrow if master tests are fixed.

**Comment by VojtechBartos** [comment]
@mkonicek updated

**Pull request closed without merging**